src/app/workflow/launch-third-party/launch-third-party.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { ga4ghPath } from '../../shared/constants';
+import { Dockstore } from '../../shared/dockstore.model';
+import { HttpParams } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LaunchThirdPartyService {
+
+  constructor() { }
+
+  public dnanexusUrl(path: string, version: string): string {
+      return Dockstore.DNANEXUS_IMPORT_URL + '?source=' + this.getTrsUrl(path, version);
+  }
+
+  public dnastackUrl(path: string, descriptorType: string): string {
+    const httpParams = new HttpParams()
+      .set('path', path)
+      .set('descriptorType', descriptorType);
+    return Dockstore.DNASTACK_IMPORT_URL + '?' + httpParams.toString();
+  }
+
+  public firecloudUrl(path: string, version: string): string {
+    return `${Dockstore.FIRECLOUD_IMPORT_URL}/${path}:${version}`;
+  }
+
+  private getTrsUrl(path: string, versionName: string): string {
```

Comment:
at some point we should move all this (and all TRS related url constructors in other files) into a separate service somewhere.